### ISSUE TITLE
Externalize some features by creating Language Server for the LSP, Language Server Protocol 

### ISSUE THREAD

**Issue by evandrocoan** [SublimeText/LaTeXTools#1277]
What do you think about it?

As the current source code is already written in Python, the LS can also be just receiving the code from here. The advantage is that our work on it would also be useful for other editors.

1. http://langserver.org
1. https://www.youtube.com/watch?v=2GqpdfIAhz8 (Language Server Protocol Explained)

**Comment by ig0774** [comment]
This project probably isn't the ideal starting point for a LaTeX LS.

* Our license is... ambiguous (see #1175) and even forking just the required components runs up against this.

* Python may not be the ideal language to write the LS in, since there's no way to guarantee that it's installed on every given client machine (Windows). Perl 5 seems a better candidate, since it's basically required for a working LaTeX install.

* Any eventual LS, if it uses the cwl completions, probably needs to be licensed under the GPL, since those files are drawn from GPL-licensed projects. We (maybe) get around this since we don't "link" to the cwl files, but simply use them if they are available. There's some discussion of this related to the Atom port in msiniscalchi/atom-latextools#53.

* Much of the way LaTeXTools is designed is somewhat dependent on it being run in ST. These may not always be the best decisions in an LS environment.

In short, LaTeXTools could be an _inspiration_ for an eventual LaTeX LS, but it's probably not anywhere near as simple as just forking (parts of) LaTeXTools and adding something to handle the LS protocol.
